### PR TITLE
Update boringtun to v0.5.2

### PR DESCRIPTION
**Pull request by arvydasv**
This only to run CI test on changes.

There are still issues with the new botringtun and dns

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean
- [x] README.md is updated
- [x] changelog.md is updated

### CODE REVIEW

**Review by gytsto — COMMENTED** [review]
+1.0

File: clis/tcli/Cargo.toml [hunk 1]
```diff
@@ -9,6 +9,8 @@ repository = "https://github.com/NordSecurity/libtelio"
 anyhow = "1"
 base64 = "0.13"
 clap = { version = "3.1", features = ["derive"] }
+clap_derive = { version = "=4.3.2" }
```
Comment (packgron) [inline clis/tcli/Cargo.toml:12]: Add comment that, these are to enforce rust 1.64 version

File: crates/telio-dns/src/dns.rs [hunk 2]
```diff
@@ -28,7 +29,7 @@ pub trait DnsResolver {
     /// Configure list of forward DNS servers for zone '.'.
     async fn forward(&self, to: &[IpAddr]) -> Result<(), String>;
     /// Get public key of this DNS server.
-    fn public_key(&self) -> PublicKey;
+    fn public_key(&self) -> PublicKeyDalek;
```
Comment (packgron) [inline crates/telio-dns/src/dns.rs:32]: Expose our PublicKey, and use `From` conversion

File: crates/telio-dns/src/dns.rs [hunk 3]
```diff
@@ -84,14 +85,10 @@ impl LocalDnsResolver {
         })?;
 
         // DNS secret key
-        let dns_secret_key = SecretKey::gen();
+        let dns_secret_key = StaticSecret::new(&mut rand::rngs::StdRng::from_entropy());
```
Comment (packgron) [inline crates/telio-dns/src/dns.rs:88]: Leave our implementation of SecretKey

File: crates/telio-dns/tests/integration_tests.rs [hunk 4]
```diff
@@ -10,35 +10,23 @@ use std::{
...
-        }
-    }
-}
+use x25519_dalek::{PublicKey, StaticSecret};
 
 /// Config is in INI format.
 struct WireguardConfig {
-    private_key: String,
-    peer_public_key: String,
+    private_key: StaticSecret,
```
Comment (packgron) [inline crates/telio-dns/tests/integration_tests.rs:17]: Always use telio_crypto definitions for our code.

**Review by packgron — CHANGES_REQUESTED** [review]

File: crates/telio-dns/tests/integration_tests.rs (hunk 4 above)
Comment (arvydasv) [reply crates/telio-dns/tests/integration_tests.rs]: Done

**Comment by dfetti** [comment]
+0.5

File: crates/telio-dns/src/dns.rs [hunk 5]
```diff
@@ -28,7 +28,7 @@ pub trait DnsResolver {
     /// Configure list of forward DNS servers for zone '.'.
     async fn forward(&self, to: &[IpAddr]) -> Result<(), String>;
     /// Get public key of this DNS server.
-    fn public_key(&self) -> PublicKey;
+    fn public_key(&self) -> [u8; 32];
```
Comment (packgron) [inline crates/telio-dns/src/dns.rs:31]: I would prefer that you would leve telio_crypto::PublicKey in public interfaces, it helps people understand what data they are working with

File: crates/telio-dns/tests/integration_tests.rs [hunk 6]
```diff
@@ -53,16 +41,19 @@ impl fmt::Display for WireguardConfig {
         write!(
             f,
             r#"[Interface] # Regular peer
-                PrivateKey = {}
+                PrivateKey = {:?}
```
Comment (packgron) [inline crates/telio-dns/tests/integration_tests.rs:44]: Why did it become debug here, display should show base64, debug concat's

Comment (arvydasv) [reply crates/telio-dns/tests/integration_tests.rs]: Done

File: crates/telio-dns/tests/integration_tests.rs [hunk 7]
```diff
@@ -58,7 +46,7 @@ impl fmt::Display for WireguardConfig {
                 ListenPort = 51820
 
                 [Peer] # DNS resolver
-                PublicKey = {}
+                PublicKey = {:?}
```
Comment (packgron) [inline crates/telio-dns/tests/integration_tests.rs:49]: This must be left as display

**Review by packgron — DISMISSED** [review]
Fix that one OK Display, othervice +1.0

**Review by dfetti — APPROVED** [review]

**Pull request merged**